src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs

```diff
@@ -109,7 +110,7 @@ public void Add<T>(T? value)
                 JsonNode? jNode = value as JsonNode;
                 if (jNode == null)
                 {
-                    jNode = new JsonValue<T>(value);
+                    jNode = new JsonValueNotTrimmable<T>(value);
```

Comment:
So adding a `T` value to a JSON array is not trim-safe. Should we add a safe version?